nitro-src/claim.ts

```diff
@@ -38,6 +44,40 @@ export function claim(
     )
       throw Error;
 
+    let surplus = BigNumber.from(initialHoldings[assetIndex]);
+    // Any guarantees before this one have priority on the funds
+    // So we must account for that by reducing the surplus
+    for (
+      let guaranteeIndex = 0;
+      guaranteeIndex < targetChannelIndex;
+      guaranteeIndex++
+    ) {
+      const { amount } = guarantees[guaranteeIndex];
+
+      surplus = surplus.sub(min(BigNumber.from(amount), surplus)); // Prevent going below 0
+    }
+    // If there are not enough funds to fund the guarantee we return immediately
+    if (surplus.lte(0)) {
```

Comment:
why use `lte(0)` rather than `isZero()`, given we prevented it going negative on L57?

Reply:
That's a good point! I think I changed these first and forgot to change them back.